massa-ledger-worker/src/ledger_db.rs

```diff
@@ -235,16 +235,33 @@ impl LedgerDB {
     /// # Returns
     /// A `BTreeSet` of the datastore keys
     pub fn get_datastore_keys(&self, addr: &Address) -> Option<BTreeSet<Vec<u8>>> {
+        let address_deserializer = AddressDeserializer::new();
         let handle = self.db.cf_handle(LEDGER_CF).expect(CF_ERROR);
 
         let mut opt = ReadOptions::default();
         opt.set_iterate_range(data_prefix!(addr).clone()..end_prefix(data_prefix!(addr)).unwrap());
 
+        /*let mut iter = self
+        .db
+        .iterator_cf_opt(handle, opt, IteratorMode::Start)
+        .flatten()
+        .map(|(key, _)| key.split_at(ADDRESS_SIZE_BYTES_V1 + 1).1.to_vec())
+        .peekable();*/
+
         let mut iter = self
             .db
             .iterator_cf_opt(handle, opt, IteratorMode::Start)
             .flatten()
-            .map(|(key, _)| key.split_at(ADDRESS_SIZE_BYTES + 1).1.to_vec())
+            .map(|(key, _)| {
+                let res: Result<(&[u8], Address), _> =
+                    address_deserializer.deserialize::<massa_serialization::DeserializeError>(&key);
+                match res {
+                    Ok((rest, _addr)) => rest.to_vec(),
+                    Err(_) => {
+                        vec![]
+                    }
+                }
+            })
```

Comment:
It seems like this will map a parsing error to an empty vec, which will be included in the btree set, thus the result being `Some(...)` instead of `None`, or a set of (serialized representation of) keys that includes an empty vec(which obfuscates the error).

one (very hacky) way to handle this, is to add a `filter(|v| !v.is_empty())`, after the `map` and before the `peekable`, but there's probably a "nicer" way to do it.